packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -214,7 +214,9 @@ class OverlayEntry implements Listenable {
   /// This method should only be called by the object's owner.
   void dispose() {
     assert(!_disposedByOwner);
-    assert(_overlay == null, 'An OverlayEntry must first be removed from the Overlay before dispose is called.');
+    // In case the owner of this Overlay
```

Comment:
this comment seems to be incomplete?

Reply:
sorry about this, I should have turn this into draft. The mounted state was because if the owner of this overlay entry is above the overlay widget. It is possible the entry will be disposed without the owner to have a chance to remove the entry. This is because the dispose starts from the child.

what the parent would know is the entry's notify the parent it has been disposed.

Things can be even worse if the entry is under another opaque entry and does not maintain state. In this case, owner won't even know the entry is disposed because it was never built

Reply:
never mind If i try really hard, i can still solve this problem